src/ast/FieldDecl.ts

```diff
@@ -12,12 +12,41 @@ export class FieldDecl extends AstNode {
 
     fields: VarList;
     modifier: string;
-    generateGetter: boolean;
-    generateSetter: boolean;
+    generateGetter: boolean = false;
+    generateSetter: boolean = false;
 
     public parse(context: Tokenizer): any {
-        // TODO: implement the rest.
+        let indentLevel: number = context.getCurrentLineTabLevel();
+        context.getAndCheckNext("fields");
+        this.modifier = context.getNext();
         this.fields = new VarList();
+        this.fields.parse(context);
+
+        context.getNext();
+
+        if (context.getCurrentLineTabLevel() <= indentLevel) {
+            return;
+        }
+
+        context.getAndCheckNext("generate");
+
+        if (context.getNext() === "getter") {
+            this.generateGetter = true;
+        } else {
+            this.generateSetter = true;
+        }
+
+        context.getNext();
+
```

Comment:
this could go after the indentationLevel check below, and could be a context.getAndCheckNext("generate") (to check whether it is "generate" or not)